### PR TITLE
Add missing Elasticsearch configuration

### PR DESCRIPTION
**Pull request by pavolloffay**
This PR adds missing Elasticserch options

All primary ES flag from Jaeger
```
      --es.bulk.actions int                                The number of requests that can be enqueued before the bulk processor decides to commit (default 1000)
      --es.bulk.flush-interval duration                    A time.Duration after which bulk requests are committed, regardless of other thresholds. Set to zero to disable. By default, this is disabled. (default 200ms)
      --es.bulk.size int                                   The number of bytes that the bulk requests can take up before the bulk processor decides to commit (default 5000000)
      --es.bulk.workers int                                The number of workers that are able to receive bulk requests and eventually commit them to Elasticsearch (default 1)
      --es.create-index-templates                          Create index templates at application startup. Set to false when templates are installed manually. (default true)
      --es.index-prefix string                             Optional prefix of Jaeger indices. For example "production" creates "production-jaeger-*".
      --es.max-num-spans int                               The maximum number of spans to fetch at a time per query in Elasticsearch (default 10000)
      --es.max-span-age duration                           The maximum lookback for spans in Elasticsearch (default 72h0m0s)
      --es.num-replicas int                                The number of replicas per index in Elasticsearch (default 1)
      --es.num-shards int                                  The number of shards per index in Elasticsearch (default 5)
      --es.password string                                 The password required by Elasticsearch
      --es.server-urls string                              The comma-separated list of Elasticsearch servers, must be full url i.e. http://localhost:9200 (default "http://127.0.0.1:9200")
      --es.sniffer                                         The sniffer config for Elasticsearch; client uses sniffing process to find all nodes automatically, disable if not required
      --es.tags-as-fields.all                              (experimental) Store all span and process tags as object fields. If true .tags-as-fields.config-file is ignored. Binary tags are always stored as nested objects.
      --es.tags-as-fields.config-file string               (experimental) Optional path to a file containing tag keys which will be stored as object fields. Each key should be on a separate line.
      --es.tags-as-fields.dot-replacement string           (experimental) The character used to replace dots (".") in tag keys stored as object fields. (default "@")
      --es.timeout duration                                Timeout used for queries. A Timeout of zero means no timeout (default 0s)
      --es.tls                                             (deprecated) see --es.tls.enabled
      --es.tls.ca string                                   Path to a TLS CA (Certification Authority) file used to verify the remote server(s) (by default will use the system truststore)
      --es.tls.cert string                                 Path to a TLS Certificate file, used to identify this process to the remote server(s)
      --es.tls.enabled                                     Enable TLS when talking to the remote server(s)
      --es.tls.key string                                  Path to a TLS Private Key file, used to identify this process to the remote server(s)
      --es.tls.server-name string                          Override the TLS server name we expect in the certificate of the remove server(s)
      --es.tls.skip-host-verify                            (insecure) Skip server's certificate chain and host name verification
      --es.token-file string                               Path to a file containing bearer token. This flag also loads CA if it is specified.
      --es.use-aliases                                     (experimental) Use read and write aliases for indices. Use this option with Elasticsearch rollover API. It requires an external component to create aliases before startup and then performing its management. Note that es.max-span-age is not taken into the account and has to be substituted by external component managing read alias.
      --es.username string                                 The username required by Elasticsearch. The basic authentication also loads CA if it is specified.
      --es.version uint                                    The major Elasticsearch version. If not specified, the value will be auto-detected from Elasticsearch.
```

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

File: pkg/exporter/elasticsearch/config.go [hunk 1]
```diff
@@ -21,6 +21,22 @@ type Config struct {
...
 	IndexPrefix string `mapstructure:"index_prefix"`
+	// Configures ES client to uses sniffing process to find all nodes automatically
+	Sniffer bool `mapstructure:"sniffer"`
+	// Use aliases for to enable Elasticsearch Rollover
+	UseWriteAlias bool `mapstructure:"use_aliases"`
+
+	Password string `mapstructure:"password"`
+	Username string `mapstructure:"username"`
+	//Path to a file containing bearer token
+	TokenFile string `mapstructure:"toke_file"`
```
Comment (objectiser) [inline pkg/exporter/elasticsearch/config.go]: Spelling `token_file`

File: pkg/exporter/elasticsearch/config.go [hunk 2]
```diff
@@ -21,6 +21,22 @@ type Config struct {
 	CreateTemplates bool `mapstructure:"create_mappings"`
 	// IndexPrefix defines options prefix of Jaeger indices. For example "production" creates "production-jaeger-*"
 	IndexPrefix string `mapstructure:"index_prefix"`
+	// Configures ES client to uses sniffing process to find all nodes automatically
```
Comment (objectiser) [inline pkg/exporter/elasticsearch/config.go]: nit: "to use sniffing"

File: pkg/exporter/elasticsearch/config.go [hunk 3]
```diff
@@ -21,6 +21,22 @@ type Config struct {
...
+	Sniffer bool `mapstructure:"sniffer"`
+	// Use aliases for to enable Elasticsearch Rollover
```
Comment (objectiser) [inline pkg/exporter/elasticsearch/config.go]: nit: remove "for"

File: pkg/exporter/elasticsearch/config.go [hunk 4]
```diff
@@ -21,6 +21,22 @@ type Config struct {
...
+	UseWriteAlias bool `mapstructure:"use_aliases"`
+
+	Password string `mapstructure:"password"`
+	Username string `mapstructure:"username"`
+	//Path to a file containing bearer token
```
Comment (objectiser) [inline pkg/exporter/elasticsearch/config.go]: nit: space between // and Path

File: pkg/exporter/elasticsearch/config.go [hunk 5]
```diff
@@ -21,6 +21,22 @@ type Config struct {
...
+
+	// Store all tags as object fields, instead nested objects
+	TagsAllAsFields bool `mapstructure:"tags_as_fields_all"`
+	// Dot replacement for tag keys when stored as object fields
+	TagsDotReplacement string `mapstructure:"tags_as_fields_dot_replacement"`
+	// File path to tag keys which should be stored as object fields
+	TagsFile string `mapstructure:"tag_as_fields_config_file"`
 
 	// BulkActions defines the number of requests that can be enqueued before the bulk processor decides to commit.
 	bulkActions int
```
Comment (objectiser) [inline pkg/exporter/elasticsearch/config.go]: Do these bulk config options need `mapstructure` as well?

Comment (pavolloffay) [reply pkg/exporter/elasticsearch/config.go]: No at the moment, we have decided to not export and see whether we could resue OTEL batch processor instead.

**Comment by pavolloffay** [comment]
> A few minor points - main thing is what is the approach with tests?

Some code has been copied over from jaeger. I will not copy over the test. Once we moce the collector to the main repository the reused symbols will be made public.

I will add configuration to existing tests.

File: pkg/exporter/elasticsearch/config.go (hunk 5 above)
Comment (objectiser) [reply pkg/exporter/elasticsearch/config.go]: ok

File: pkg/exporter/elasticsearch/config.go [hunk 6]
```diff
@@ -21,6 +21,17 @@ type Config struct {
...
+	Sniffer bool `mapstructure:"sniffer"`
+	// Use aliases to enable Elasticsearch Rollover
+	UseWriteAlias bool `mapstructure:"use_aliases"`
+
+	Password string `mapstructure:"password"`
+	Username string `mapstructure:"username"`
+	// Path to a file containing bearer token
+	TokenFile string `mapstructure:"token_file"`
+
+	TagsAsFields TagsAsFields `mapstructure:"tags_as_fields"`
```
Comment (pavolloffay) [inline pkg/exporter/elasticsearch/config.go]: Moving this to a nested struct to better align with the current format

**Pull request merged**